Mathlib/Analysis/Calculus/FDeriv/Prod.lean

```diff
@@ -355,6 +355,25 @@ alias DifferentiableAt.prod_map := DifferentiableAt.prodMap
 
 end prodMap
 
+section Swap
```

Comment:
I think these lemmas are useful to have - and could be split into its own PR. (I'd be happy to review that one.)
I certainly think that some versions without `Within` should also be proven (`hasFDerivAt`and `fderiv_swap` for sure, for instance) --- perhaps all the lemmas you mention below are useful.